packages/x-charts/src/internals/useAnimate.ts

```diff
@@ -0,0 +1,123 @@
+'use client';
+import useId from '@mui/utils/useId';
+import * as React from 'react';
+import { interrupt, Transition } from '@mui/x-charts-vendor/d3-transition';
+import { select } from '@mui/x-charts-vendor/d3-selection';
+import { shallowEqual } from './shallowEqual';
+import { ANIMATION_DURATION_MS, ANIMATION_TIMING_FUNCTION_JS } from '../constants';
+import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect';
+
+export function useAnimate<Props extends {}, Elem extends Element>(
```

Comment:
This probably needs to accept a ref if we want to allow public use 😆

Since we return a ref, users that need a ref themselves wouldn't be able to add a ref to their element.

Reply:
`useAnimate` won't be for public use. However, I agree that we should probably do that for the `useAnimatedLine` or `useAnimatedPieArc` hooks that will be exported.